src/topology/algebra/group.lean

```diff
@@ -746,3 +746,100 @@ instance : topological_group (units α) :=
+⟨{to_topological_space := ⊤,
+  continuous_mul       := continuous_top,
+  continuous_inv       := continuous_top}⟩
+
+variables {γ : Type*} [group γ]
+
+@[ext]
+lemma ext' {f g : group_topology γ} (h : f.is_open = g.is_open) : f = g :=
+by { ext, rw h }
+
+/-- The ordering on group topologies on the group `γ`.
+  `t ≤ s` if every set open in `s` is also open in `t` (`t` is finer than `s`). -/
+instance : partial_order (group_topology γ) :=
+partial_order.lift group_topology.to_topological_space $ ext
+
+local notation `cont` := @continuous _ _
+
+private def def_Inf (S : set (group_topology γ)) : group_topology γ :=
+let Inf_S' := Inf (to_topological_space '' S) in
+{ to_topological_space := Inf_S',
+  continuous_mul       :=
+  begin
+    apply continuous_Inf_rng,
+    rintros _ ⟨⟨t, tr⟩, haS, rfl⟩, resetI,
+    have h := continuous_Inf_dom (set.mem_image_of_mem to_topological_space haS) continuous_id,
+    have h_continuous_id := @continuous.prod_map _ _ _ _ t t Inf_S' Inf_S' _ _ h h,
+    have h_continuous_mul : cont (id _) t (λ (p : γ × γ), p.fst * p.snd) := continuous_mul,
+    exact @continuous.comp _ _ _ (id _) (id _) t _ _ h_continuous_mul h_continuous_id,
+  end,
+  continuous_inv       :=
+  begin
+    apply continuous_Inf_rng,
+    rintros _ ⟨⟨t, tr⟩, haS, rfl⟩, resetI,
+    exact @continuous.comp _ _ _ (id _) (id _) t _ _ continuous_inv
+      (continuous_Inf_dom (set.mem_image_of_mem to_topological_space haS) continuous_id),
+  end, }
+
+/-- Group topologies on `γ` form a complete lattice, with `⊥` the discrete topology and `⊤` the
+indiscrete topology.
+
+The infimum of a collection of group topologies is the topology generated by all their open sets
+(which is a group topology).
+
+The supremum of two group topologies `s` and `t` is the infimum of the family of all group
+topologies contained in the intersection of `s` and `t`. -/
+instance : complete_semilattice_Inf (group_topology γ) :=
+{ Inf    := def_Inf,
+  Inf_le := λ S a haS, by { apply topological_space.complete_lattice.Inf_le, use [a, ⟨ haS, rfl⟩] },
+  le_Inf :=
+  begin
+    intros S a hab,
+    apply topological_space.complete_lattice.le_Inf,
+    rintros _ ⟨b, hbS, rfl⟩,
+    exact hab b hbS,
+  end,
+  ..group_topology.partial_order }
+
+instance : complete_lattice (group_topology γ) :=
+complete_lattice_of_complete_semilattice_Inf _
+
+/--  Given `f : α → β` and a topology on `α`, the coinduced group topology on `β` is the finest
+topology such that `f` is continuous and `β` is a topological group. -/
+def coinduced {α β : Type*} [t : topological_space α] [group β] (f : α → β) :
+  group_topology β :=
+Inf {b : group_topology β | (topological_space.coinduced f t) ≤ b.to_topological_space}
+
+lemma coinduced_continuous {α β : Type*} [t : topological_space α] [group β] (f : α → β) :
+  cont t (coinduced f).to_topological_space f :=
+begin
+  rw continuous_iff_coinduced_le,
+  refine le_Inf _,
+  rintros _ ⟨t', ht', rfl⟩,
+  exact ht',
+end
+
+
```

Comment:
In the file on `ring_topology`, could you please provide the order embedding from `ring_topology` to `add_group_topology` (once the latter exists).

Reply:
OK, I'll write that.